### DISCUSSION TITLE
JSVM hook streaming support

### DISCUSSION THREAD

**Discussion by lestan** [pocketbase/pocketbase#6135] [Q&A]
Hoping to get some help with a use case that I can't seem to figure out.

I'm extending PB with a server-side JS hook.

Within the request router, I'm making a HTTP call using the $http.send() method.  The endpoint is responding with a stream.  I'd like to receive chunks of the response and stream it back to the web client.

I'm assuming I have to use e.stream() to stream the response back.  But unsure how to get the streaming response from $http.send()

[![](https://mermaid.ink/img/pako:eNplkVFLwzAUhf9KCAiKKxUfA264iSBMLO5BkILcpcc2tE1qkk7G2H83tV1XMQ8h9-bku_fcHLg0GbjgDl8ttMSDotxSnWoW1hu2q0pB-2g-v06Wgm2gM2Y7qfPsMpYF-TiHhiWPRcja_d13yF3c3igXtsbIEn5LDournpgsO9R6_fxxnzwJ9tJAM-ctqFY6P5ML7xsRx1VVO9gd7Fik1w6wARMFZDQif1vckiyZLFpdurHuycMrJNQO_TU6O64x2uGPcHQ-AINY_hdPGujIj0pT1XMnbqMJa1UZh8Ewn_EatiaVhfEfugcp9wVqpFyEY0a2THmqj0FHrTebvZZceNtixq1p84KLT6pciNomC5MZPm7MNqTfjTnFxx9SJKFj?type=png)](https://mermaid.live/edit#pako:eNplkVFLwzAUhf9KCAiKKxUfA264iSBMLO5BkILcpcc2tE1qkk7G2H83tV1XMQ8h9-bku_fcHLg0GbjgDl8ttMSDotxSnWoW1hu2q0pB-2g-v06Wgm2gM2Y7qfPsMpYF-TiHhiWPRcja_d13yF3c3igXtsbIEn5LDournpgsO9R6_fxxnzwJ9tJAM-ctqFY6P5ML7xsRx1VVO9gd7Fik1w6wARMFZDQif1vckiyZLFpdurHuycMrJNQO_TU6O64x2uGPcHQ-AINY_hdPGujIj0pT1XMnbqMJa1UZh8Ewn_EatiaVhfEfugcp9wVqpFyEY0a2THmqj0FHrTebvZZceNtixq1p84KLT6pciNomC5MZPm7MNqTfjTnFxx9SJKFj)

I hope this makes sense!  Appreciate any guidance.

**Comment by ganigeorgiev** [comment]
This was asked several times in the past but there is no support at the moment for streamed responses with `$http.send` and you'll have to use Go (_I'll add a "Limitations" section to the `http.send` docs later today to make it more clear_).

Note that what you want is probably an SSE client (so that you could parse the complete chunked message, to handle reconnects, etc.) but for now I'm not planning working on this for the JSVM as I already have enough on my hands and such use cases usually are more suitable for Go due to its builtin concurrency primitives.
